drivers/counter/counter_nrfx_rtc.c

```diff
@@ -654,18 +654,25 @@ static const struct counter_driver_api counter_nrfx_driver_api = {
 	.set_guard_period = set_guard_period,
 };
 
-#define COUNTER_NRF_RTC_DEVICE(idx)					\
-	BUILD_ASSERT((DT_NORDIC_NRF_RTC_RTC_##idx##_PRESCALER - 1) <=	\
-		     RTC_PRESCALER_PRESCALER_Msk,			\
-		     "RTC prescaler out of range");			\
+/*
+ * Devicetree access is done with node labels due to HAL API
+ * requirements. In particular, RTCx_CC_NUM values from HALs
+ * are indexed by peripheral number, so DT_INST APIs won't work.
+ */
+
+#define RTC(idx)		DT_NODELABEL(rtc##idx)
+#define RTC_PROP(idx, prop)	DT_PROP(RTC(idx), prop)
+
+#define COUNTER_NRF_RTC_DEVICE(idx)					       \
+	BUILD_ASSERT((RTC_PROP(idx, prescaler) - 1) <=			       \
+		     RTC_PRESCALER_PRESCALER_Msk,			       \
+		     "RTC prescaler out of range");			       \
```

Comment:
What about including `idx` in the message, so that there's no need to go through the macro expansion trace to get it?

Reply:
@mbolivar-nordic ^^